docs/proposals/30-apiserver-proxy.md

```diff
@@ -0,0 +1,317 @@
+...
+```
+
+This solution would be reusing existing Envoy filtering on the `apiserver-proxy` pod and simply exchange its Proxy Protocol configuration for the new proposed header.
+
+### Istio IngressGateway Configuration
+
+Similarly, it would be required to configure the Istio IngressGateway to accept HTTP CONNECT requests and route them appropriately. This could e.g. be achieved utilizing Istio Custom Resources (Gateway, VirtualService):
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: api-gateway
+  namespace: istio-system
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+    ...
+    ...
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: api-routes
+  namespace: istio-system
+spec:
+  ...
+  ...
+  gateways:
+  - api-gateway
+  http:
+  - match:
+    - method:
+        exact: CONNECT
+      headers:
+        X-Gardener-API-Route:
+          regex: ".*"
+    ...
+    ...
+```
+
+This configuration sets up the Istio IngressGateway to accept HTTPS traffic and route HTTP CONNECT requests with the proposed custom header to the `kube-apiserver`.
+
+### EnvoyFilter for Custom Header Processing
+
+To process the proposed custom header and make routing decisions based on it, it will be required to additionally add and/or modify an EnvoyFilter. Again, existing configuration may be reused adjusted to this new use case to keep convention parity within the code base.
+
+```yaml
+{{- if eq .Values.vpn.enabled true -}}
+apiVersion: networking.istio.io/v1beta1
+kind: Gateway
+metadata:
+  name: reversed-vpn-auth-server
+  namespace: {{ .Release.Namespace }}
+spec:
+  selector:
+{{ .Values.labels | toYaml | indent 4 }}
+  servers:
+  - hosts:
+    - reversed-vpn-auth-server.garden.svc.cluster.local
+    port:
+      name: tls-tunnel
+      number: 8134
+      protocol: HTTP
+{{- end }}
+```
+### Implementation Steps
+
+1. Apply the Istio Gateway and VirtualService configurations.
+2. Deploy the EnvoyFilter for custom header processing.
+3. Update any relevant Istio authorization policies to allow the new traffic flow.
+4. Update the `apiserver-proxy` code or configuration to use HTTP CONNECT and include the custom header.
+5. Deploy the updated `apiserver-proxy` configuration.
+
+## Phased Rollout
+
+The implementation will be controlled by two distinct feature gates that handle different aspects of the solution. Each feature gate can be disabled per shoot to allow testing of the old implementation via E2E tests.
+
+### Feature Gate 1: `APIServerSecureRouting`
+Controls the enablement of the new routing implementation using HTTP CONNECT and custom headers.
+
+#### Alpha
+- **Default:** Disabled
+- **Actions:**
+    - Introduces new port for HTTP CONNECT
+    - Deploys new custom header processing
+    - Configures mTLS for new routing path
+    - Enables E2E testing of the feature
+    - Implementation of all unit and integration tests
+    - Documentation completion
```

Comment:
I do not like this documentation. Obviously, first things need to be implemented, tested and documented, but this has nothing to do with the state of the feature gate. Why do you make it look like there is a direct correlation with the feature gate status?